src/main/webapp/js/projects/react/components/ExpandedProject.js

```diff
@@ -0,0 +1,28 @@
+import {expandedProjectType} from '../prop_types.js';
+
+/**
+ * An expanded view of a project.
+ * @param {Object} props
+ * @return {React.Component} Returns the React component.
+ */
+export default function ExpandedProject(props) {
+  if (props.loading) {
```

Comment:
Very interesting stuff. When does this check for the loading property get refreshed?

Reply:
Good question! When the `DataFetcher` component finishes fetching the data, it will update its state using `this.setState([...])`. When the state is updated this way, it [triggers a re-render](https://reactjs.org/docs/react-component.html#setstate) of the component:

> setState() enqueues changes to the component state and tells React that this component and its children need to be re-rendered with the updated state.

Once the `DataFetcher` component is re-rendered, it will also re-render the `ExpandedProject` component, which will then re-run the function that you see above.

Reply:
I see, the whole cascading thing. very cool!